### ISSUE TITLE
bytes.concat is not supported

### ISSUE THREAD

**Issue by axic** [crytic/slither#944]
`bytes.concat` was introduced in 0.8.4.

```
contract A {
  function f() external returns (bytes memory) {
    return bytes.concat();
  }
}
```

will result in
```
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.9/site-packages/slither/__main__.py", line 741, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/__main__.py", line 83, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/__main__.py", line 66, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/opt/homebrew/lib/python3.9/site-packages/slither/slither.py", line 107, in __init__
    parser.analyze_contracts()
  File "/opt/homebrew/lib/python3.9/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 393, in analyze_contracts
    self._convert_to_slithir()
  File "/opt/homebrew/lib/python3.9/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 570, in _convert_to_slithir
    func.generate_slithir_and_analyze()
  File "/opt/homebrew/lib/python3.9/site-packages/slither/core/declarations/function.py", line 1676, in generate_slithir_and_analyze
    node.slithir_generation()
  File "/opt/homebrew/lib/python3.9/site-packages/slither/core/cfg/node.py", line 712, in slithir_generation
    self._irs = convert_expression(expression, self)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/slithir/convert.py", line 112, in convert_expression
    visitor = ExpressionToSlithIR(expression, node)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/visitors/slithir/expression_to_slithir.py", line 129, in __init__
    self._visit_expression(self.expression)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/visitors/expression/expression.py", line 51, in _visit_expression
    self._visit_call_expression(expression)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/visitors/expression/expression.py", line 108, in _visit_call_expression
    self._visit_expression(expression.called)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/visitors/expression/expression.py", line 95, in _visit_expression
    self._post_visit(expression)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/visitors/expression/expression.py", line 289, in _post_visit
    self._post_member_access(expression)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/visitors/slithir/expression_to_slithir.py", line 389, in _post_member_access
    member = Member(expr, Constant(expression.member_name), val)
  File "/opt/homebrew/lib/python3.9/site-packages/slither/slithir/operations/member.py", line 23, in __init__
    assert is_valid_rvalue(variable_left) or isinstance(
AssertionError
None
Error in .
```

**Comment by axic** [comment]
@montyly this causes a crash, so it prevents running slither on certain newer codebases.

**Comment by 0xalpharush** [comment]
Fixed by #985
